vector/v.colors/scan_attr.c

```diff
@@ -42,6 +42,7 @@ int scan_attr(struct Map_info *Map, int layer, const char *column_name,
                                  &cvarr);
     if (nrec < 1) {
         G_important_message(_("No data selected"));
+        Vect_destroy_field_info(fi);
```

Comment:
Here you can also close the database (`driver`) before return.

Reply:
done